cmd/query/app/querysvc/query_service.go

```diff
@@ -166,6 +169,26 @@ func (opts *QueryServiceOptions) InitArchiveStorage(storageFactory storage.Facto
 	return true
 }
 
+// InitMetricStorage tries to initialize metric storage reader if storage factory supports them.
+func (opts *QueryServiceOptions) InitMetricStorage(metricsFactory storage.MetricsFactory, logger *zap.Logger) (metricsstore.Reader, bool) {
```

Comment:
What does this function have to do with `QueryServiceOptions`?  All you do is call the factory, this can be done directly in the `queryservice` extension, similar to how it creates span storage components.

Reply:
Understood. I have moved this to `createMetricStorage` in the query extension.